cleantalk.php

```diff
@@ -638,6 +638,23 @@ function apbct_sfw__check()
         return;
     }
 
+    // Checking if database was outdated
+    if ( $apbct->stats['sfw']['last_update_time'] + $apbct->stats['sfw']['update_period'] * 3 < time() ) {
```

Comment:
There is a method State::errorToggle() it could perfectly fit here.

Reply:
State::errorToggle() method does not check the existence of the error and every time rewrite this. It is not good for the performance.

Reply:
You could extend the method to use it this way.

Reply:
Done https://github.com/CleanTalk/wordpress-antispam/pull/40/commits/4d7dc2151e32e65b1bd083ec9221cbdd8ddc7767